addons/mail/static/src/components/message/message.xml

```diff
@@ -81,6 +81,25 @@
                                         Anonymous
                                     </div>
                                 </t>
+                                <t t-if="threadView and messageView.message.originThread and messageView.message.originThread === threadView.thread and messageView.message.notifications.length > 0">
+                                    <t t-if="messageView.message.failureNotifications.length > 0">
+                                        <span class="o_Message_notificationIconClickable o-error mr-1" role="button" tabindex="0" t-on-click="_onClickFailure">
+                                            <i name="failureIcon" class="o_Message_notificationIcon fa fa-envelope" role="img" aria-label="Delivery failure"/>
+                                        </span>
+                                    </t>
+                                    <t t-else="">
+                                        <Popover>
+                                            <span class="o_Message_notificationIconClickable mr-1">
+                                                <i name="notificationIcon" class="o_Message_notificationIcon fa fa-envelope-o"/>
+                                            </span>
+                                            <t t-set="opened">
+                                                <NotificationPopover
+                                                    notificationLocalIds="messageView.message.notifications.map(notification => notification.localId)"
+                                                />
+                                            </t>
+                                        </Popover>
+                                    </t>
+                                </t>
```

Comment:
I added `mr-1` because it looks off for `- datetime` to be touching the envelope icon